src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CancellationTokenParametersMustComeLast.cs

```diff
@@ -99,6 +100,19 @@ public override void Initialize(AnalysisContext context)
                         last--;
                     }
 
+                    // Ignore parameters that hav any of these attributes.
+                    // C# reserved attributes: https://docs.microsoft.com/dotnet/csharp/language-reference/attributes/caller-information
+                    var callerInformationAttributes = ImmutableArray.Create<INamedTypeSymbol>(
+                        compilationContext.Compilation.GetTypeByMetadataName(typeof(CallerFilePathAttribute).FullName),
+                        compilationContext.Compilation.GetTypeByMetadataName(typeof(CallerLineNumberAttribute).FullName),
+                        compilationContext.Compilation.GetTypeByMetadataName(typeof(CallerMemberNameAttribute).FullName));
```

Comment:
You are using the wrong API.

```suggestion
                        compilationContext.Compilation.GetOrCreateTypeByMetadataName(typeof(CallerFilePathAttribute).FullName),
                        compilationContext.Compilation.GetOrCreateTypeByMetadataName(typeof(CallerLineNumberAttribute).FullName),
                        compilationContext.Compilation.GetOrCreateTypeByMetadataName(typeof(CallerMemberNameAttribute).FullName));
```

Reply:
@Evangelink Yeah looks like Compilation.GetTypeByMetadataName is even banned here.
Do you know why? I used it before in a personal source generator and it's used heavily in dotnet/roslyn without issues.

Reply:
Yes I know why :) Basically the resolution done within the `GetTypeByMetadataName` is expensive so we are building a cache to ensure the expensive part is done only once per compilation.